packages/examples/cvat/exchange-oracle/src/handlers/escrow_cleanup.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+import contextlib
+from typing import TYPE_CHECKING
+
+from cvat_sdk.api_client.exceptions import NotFoundException
+
+import src.cvat.api_calls as cvat_api
+import src.services.cloud as cloud_service
+from src.core.config import Config
+from src.core.storage import (
+    compose_data_bucket_prefix,
+    compose_results_bucket_prefix,
+)
+from src.services.cloud.utils import BucketAccessInfo
+
+if TYPE_CHECKING:
+    from src.models.cvat import Project
+
+
+def _cleanup_cvat(projects: list[Project]) -> None:
+    deleted_cloud_storage_ids = set()
+    for project in projects:
+        if (
+            project.cvat_cloudstorage_id is not None
+            and project.cvat_cloudstorage_id not in deleted_cloud_storage_ids
+        ):
+            with contextlib.suppress(NotFoundException):
+                # probably will allways call this just once
+                cvat_api.delete_cloudstorage(project.cvat_cloudstorage_id)
+            deleted_cloud_storage_ids.add(project.cvat_cloudstorage_id)
```

Comment:
Probably, it's better to reorder calls here, so that CS ids are collected first, then removed. Otherwise it's possible that a project depending on a CS in CVAT has no CS link. Probably, it should be controlled by CVAT and a 400 error should be raised on such a removal attempt. A CS should only be removed when there are no projects using it.